controllers/report_controller.go

```diff
@@ -70,8 +76,49 @@ func (r *ReportReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		l.Error(err, "failed to ensure the report")
 		return ctrl.Result{}, err
 	}
+	now := time.Now().UTC()
+	logrLogger, err := logr.FromContext(ctx)
+	if err != nil {
+		return ctrl.Result{}, nil
+	}
+	reportPeriod, err := getReportPeriod(now, logrLogger, report)
+	if err != nil {
+		return ctrl.Result{}, nil
+	}
+	//l.Info("reconciling report", "periodStart", reportPeriod.periodStart)
+	//l.Info("reconciling report", "now", now)
+	//l.Info("reconciling report", "periodEnd", reportPeriod.periodEnd)
+	//l.Info("reconciling report", "waits for", reportPeriod.periodEnd.Sub(now))
```

Comment:
nit: this is probably fine to log but at a higher log verbosity than the info level.